sdk/core/Azure.Core/src/Pipeline/DefaultRetryPolicy.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Core.Pipeline
+{
+    internal class DefaultRetryPolicy : RetryPolicy
```

Comment:
Why do we need this type? Why not use the RetryPolicy?

Reply:
This is an internal concrete implementation of the abstract policy.

Reply:
Will consider just making RetryPolicy concrete instead.

Reply:
For now I will leave it as abstract. We can discuss further during API review.